airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -287,6 +291,14 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
             throw new SourceException("Source process read attempt failed", e);
           }
           if (messageOptional.isPresent()) {
+            if (messageOptional.get().getRecord() != null) {
+              try {
+                recordSchemaValidator.validateSchema(messageOptional.get().getRecord());
+              } catch (final RecordSchemaValidationException e) {
+                LOGGER.warn(e.getMessage());
```

Comment:
@alovew is this printing a warning message for each record during a sync (in case of bad validation)? 

In case of a large sync where all records are badly formatted the same way, this would fill up the logs with tons of similar warning if i am not wrong?

See reports from users here: https://github.com/airbytehq/airbyte/issues/12804